### ISSUE TITLE
Ingestion into existing table doesn't check Dataframe's schema matching

### ISSUE THREAD

**Issue by DashaBulanova** [Azure/azure-kusto-spark#186]
Hello Guys
**Describe the bug**
In ingestion process If dataframe schema doesn't match an existing table in case either in column count or in column order, the KustoClient doesn't raise an exception and ingests values into wrong columns.

**To Reproduce**
1. Supposed  we have a table:
`.create table KustoSparkClient (Long: int, String: string) `

2. Then ingest row in one columns order:
```
val data = Seq((110, "my-row-110"))
val dfOrig = data.toDF( "Long", "String")
dfOrig.write
      .format("com.microsoft.kusto.spark.datasource")
      ... //kusto options here
      .mode(SaveMode.Append)
      .save()
```
3. Then ingest another row and just switch the order of columns:
```
val data = Seq(("my-row-110", 110,))
val dfOrig = data.toDF( "String", "Long")
dfOrig.write
      .format("com.microsoft.kusto.spark.datasource")
      ... //kusto options here
      .option("tableCreateOptions", "CreateIfNotExist")
      .mode(SaveMode.Append)
      .save()
```

**Expected behavior**
I expect that my row from the second step will pass into the correct column in kusto or mismatch exception but see strange behavior (see screenshot):

![image](https://user-images.githubusercontent.com/948846/120339332-b25ebe00-c2fd-11eb-8b07-1bc54ed20bfe.png)

Also, we found a comment in the code that looks like it would like to check matches but doesn't do it.

https://github.com/Azure/azure-kusto-spark/blob/59eabb68183cf82d1d7fd52f684c20dc0f66201f/connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala#L72

**Comment by ohadbitt** [comment]
Hi 
This isn't something the connector currently does - it is something nice to have - i will add an option for it
To unblock - you should use a ingestion mapping option - specifying a csvMapping as the connector produce csv files
https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/mappings
https://github.com/Azure/azure-kusto-spark/blob/master/docs/KustoSink.md

**Comment by ohadbitt** [comment]
Discussed privately and a contribution was made 
https://github.com/Azure/azure-kusto-spark/pull/196
:)
